pkg/elementalConfig/config.go

```diff
@@ -42,83 +39,62 @@ import (
 	"k8s.io/mount-utils"
 )
 
-type GenericOptions func(a *v1.Config) error
+type GenericOptions func(a *v1.Config)
 
-func WithFs(fs v1.FS) func(r *v1.Config) error {
-	return func(r *v1.Config) error {
+func WithFs(fs v1.FS) func(r *v1.Config) {
+	return func(r *v1.Config) {
 		r.Fs = fs
-		return nil
 	}
 }
 
-func WithLogger(logger v1.Logger) func(r *v1.Config) error {
-	return func(r *v1.Config) error {
+func WithLogger(logger v1.Logger) func(r *v1.Config) {
+	return func(r *v1.Config) {
 		r.Logger = logger
-		return nil
 	}
 }
 
-func WithSyscall(syscall v1.SyscallInterface) func(r *v1.Config) error {
-	return func(r *v1.Config) error {
+func WithSyscall(syscall v1.SyscallInterface) func(r *v1.Config) {
+	return func(r *v1.Config) {
 		r.Syscall = syscall
-		return nil
 	}
 }
 
-func WithMounter(mounter mount.Interface) func(r *v1.Config) error {
-	return func(r *v1.Config) error {
+func WithMounter(mounter mount.Interface) func(r *v1.Config) {
+	return func(r *v1.Config) {
 		r.Mounter = mounter
-		return nil
 	}
 }
 
-func WithRunner(runner v1.Runner) func(r *v1.Config) error {
-	return func(r *v1.Config) error {
+func WithRunner(runner v1.Runner) func(r *v1.Config) {
+	return func(r *v1.Config) {
 		r.Runner = runner
-		return nil
 	}
 }
 
-func WithClient(client v1.HTTPClient) func(r *v1.Config) error {
-	return func(r *v1.Config) error {
+func WithClient(client v1.HTTPClient) func(r *v1.Config) {
+	return func(r *v1.Config) {
 		r.Client = client
-		return nil
 	}
 }
 
-func WithCloudInitRunner(ci v1.CloudInitRunner) func(r *v1.Config) error {
-	return func(r *v1.Config) error {
+func WithCloudInitRunner(ci v1.CloudInitRunner) func(r *v1.Config) {
+	return func(r *v1.Config) {
 		r.CloudInitRunner = ci
-		return nil
 	}
 }
 
-func WithArch(arch string) func(r *v1.Config) error {
-	return func(r *v1.Config) error {
-		r.Arch = arch
-		return nil
-	}
-}
-
-func WithPlatform(platform string) func(r *v1.Config) error {
-	return func(r *v1.Config) error {
+func WithPlatform(platform string) func(r *v1.Config) {
```

Comment:
Not sure I understand why we can get rid of the error

Reply:
Mainly because we dont use it anywhere in a proper way so its useless. This functions are just for setting private values on the struct and that should never fail.